DisCatSharp.ApplicationCommands/Enums/ApplicationCommandModuleLifespan.cs

```diff
@@ -0,0 +1,44 @@
+// This file is part of the DisCatSharp project, based off DSharpPlus.
+//
+// Copyright (c) 2021-2022 AITSYS
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+namespace DisCatSharp.ApplicationCommands.Enums;
+
+/// <summary>
+/// Represents a application command module lifespan.
+/// </summary>
+public enum ApplicationCommandModuleLifespan
+{
+	/// <summary>
+	/// Whether this module should be initiated every time a command is run, with dependencies injected from a scope.
+	/// </summary>
+	Scoped,
+
+	/// <summary>
+	/// Whether this module should be initiated every time a command is run.
+	/// </summary>
+	Transient,
+
+	/// <summary>
+	/// Whether this module should be initiated at startup.
+	/// </summary>
+	Singleton
+}
```

Comment:
This is already a thing that exists within C#'s Dependency Injection module. No need to reinvent the wheel here

```cs
using Microsoft.Extensions.DependencyInjection;
```
ServiceLifetime - which has these things you defined

Reply:
:o